src/Simplex/Chat/Store/NoteFolders.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Simplex.Chat.Store.NoteFolders where
+
+import Control.Monad.Except (ExceptT (..))
+import Control.Monad.IO.Class (liftIO)
+import Data.Time (getCurrentTime)
+import Database.SQLite.Simple (Only (..))
+import Database.SQLite.Simple.QQ (sql)
+import Simplex.Chat.Store.Shared (StoreError (..), checkConstraint)
+import Simplex.Chat.Types (NoteFolder (..), NoteFolderId, NoteFolderName, User (..))
+import Simplex.Messaging.Agent.Protocol (UserId)
+import Simplex.Messaging.Agent.Store.SQLite (firstRow)
+import qualified Simplex.Messaging.Agent.Store.SQLite.DB as DB
+
+createNoteFolder :: DB.Connection -> User -> ExceptT StoreError IO ()
+createNoteFolder db User {userId} =
+  checkConstraint SENoteFolderAlreadyCreated . liftIO $
+    DB.execute db [sql| INSERT INTO note_folders (user_id) VALUES (?) |] (Only userId)
+
+getNoteFolderIdByName :: DB.Connection -> User -> NoteFolderName -> ExceptT StoreError IO NoteFolderId
+getNoteFolderIdByName db User {userId} ldn =
+  ExceptT . firstRow fromOnly (SENoteFolderNotFoundByName ldn) $
+    DB.query db [sql| SELECT note_folder_id FROM note_folders WHERE user_id = ? AND "-" = ? |] (userId, ldn)
```

Comment:
```suggestion
    DB.query db "SELECT note_folder_id FROM note_folders WHERE user_id = ? AND "-" = ?" (userId, ldn)
```

Reply:
what's this filter? `"-" = ?`

Reply:
actually the whole function doesn't seem to make sense right now as they don't have name

Reply:
They have have the name, it just happens to be empty. I left the check specifically to prevent future confusion when non-empty names would be added.